java/google/registry/flows/domain/ExtraDomainValidation.java

```diff
@@ -0,0 +1,41 @@
+package google.registry.flows.domain;
+
+import org.joda.time.DateTime;
+
+import google.registry.flows.EppException;
+import google.registry.model.domain.launch.LaunchCreateExtension;
+import google.registry.model.registry.label.ReservationType;
+
+/**
+ * Provides an injection point for custom Domain name validation during flows
+ */
+public interface ExtraDomainValidation {
+
+  /**
+   * An Injection point for domain create logic
+   *
+   * @param label the second level domain name string
+   * @param tld the top level domain name string
+   * @param launchCreate an epp create extension
+   * @param now the date time this flow was executed
+   * @throws EppException if the custom validation fails
+   */
+  void validateDomainCreate(final String label,
```

Comment:
Am I correct that your basic idea is to have a single injected ExtraDomainValidation implementation in the system, which checks all TLDs? My concern is that if you implement an extra validation check, and then you want another one, and another one, you'd have to stick all the checks together in one ExtraDomainValidation object, even when they are logically separate. For instance, if you implement a check to keep people from registering a name beginning with A on Mondays and Tuesdays, and then you want to add a check to make sure that all domain names have an odd number of letters, those would have to go together in the same check. It seems like it would be cleaner to allow more than one extra validator somehow. Then you could just configure as many independent validators as you wanted. Does that make sense?

Also, the check method knows only about the domain name, launch extension and as-of date. Wouldn't we sometimes also want to know who the registrar is? Maybe not for the exact check you have in mind at first, but eventually there will surely be registrar-specific constraints. It seems wise to build in a way for the checker to have access to additional information. Maybe just pass the entire EppInput object, and let the checker extract what it needs.